[mimalloc] build failure

**Host Environment**

- OS: Windows]
- Compiler: visual studio 2019

**To Reproduce**

Steps to reproduce the behavior:

`.\vcpkg.exe install mimalloc`

**Failure logs**
```
The following packages will be built and installed:
    mimalloc[core]:x86-windows -> 2.0.9
Detecting compiler hash for triplet x86-windows...
-- Automatically setting HTTP(S)_PROXY environment variables to "127.0.0.1:1080".
Restored 0 package(s) from C:\Users\70918\AppData\Local\vcpkg\archives in 115.5 us. Use --debug to see more details.
Installing 1/1 mimalloc:x86-windows...
Building mimalloc[core]:x86-windows...
-- Using cached microsoft-mimalloc-v2.0.9.tar.gz.
-- Cleaning sources at E:/thirdsource/vcpkg/buildtrees/mimalloc/src/v2.0.9-b57e26b15c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/thirdsource/vcpkg/downloads/microsoft-mimalloc-v2.0.9.tar.gz
-- Applying patch fix-cmake.patch
-- Using source at E:/thirdsource/vcpkg/buildtrees/mimalloc/src/v2.0.9-b57e26b15c.clean
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "D:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j21
    Working Directory: E:/thirdsource/vcpkg/buildtrees/mimalloc/x86-windows-dbg
    See logs for more information:
      E:\thirdsource\vcpkg\buildtrees\mimalloc\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/mimalloc/portfile.cmake:36 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)
```
**mimalloc\install-x86-windows-dbg-out.log**
```
[1/16] D:\PROGRA~1\MICROS~1\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x86\cl.exe   /TP -DMI_DEBUG=3 -DMI_SHARED_LIB -DMI_SHARED_LIB_EXPORT -Dmimalloc_EXPORTS -IE:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /Zc:__cplusplus -std:c++17 /showIncludes /FoCMakeFiles\mimalloc.dir\src\alloc-posix.c.obj /FdCMakeFiles\mimalloc.dir\ /FS -c E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\src\alloc-posix.c
FAILED: CMakeFiles/mimalloc.dir/src/alloc-posix.c.obj 
D:\PROGRA~1\MICROS~1\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x86\cl.exe   /TP -DMI_DEBUG=3 -DMI_SHARED_LIB -DMI_SHARED_LIB_EXPORT -Dmimalloc_EXPORTS -IE:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /Zc:__cplusplus -std:c++17 /showIncludes /FoCMakeFiles\mimalloc.dir\src\alloc-posix.c.obj /FdCMakeFiles\mimalloc.dir\ /FS -c E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\src\alloc-posix.c
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(503): error C2975: 'destroy': invalid template argument for '_mi_heap_stl_allocator_common', expected compile-time constant expression
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(484): note: see declaration of 'destroy'
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(519): note: see reference to class template instantiation '_mi_heap_stl_allocator_common<T,destroy>' being compiled
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(514): error C2975: 'destroy': invalid template argument for '_mi_heap_stl_allocator_common', expected compile-time constant expression
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(484): note: see declaration of 'destroy'
[2/16] D:\PROGRA~1\MICROS~1\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x86\cl.exe   /TP -DMI_DEBUG=3 -DMI_SHARED_LIB -DMI_SHARED_LIB_EXPORT -Dmimalloc_EXPORTS -IE:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /Zc:__cplusplus -std:c++17 /showIncludes /FoCMakeFiles\mimalloc.dir\src\stats.c.obj /FdCMakeFiles\mimalloc.dir\ /FS -c E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\src\stats.c
FAILED: CMakeFiles/mimalloc.dir/src/stats.c.obj 
D:\PROGRA~1\MICROS~1\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x86\cl.exe   /TP -DMI_DEBUG=3 -DMI_SHARED_LIB -DMI_SHARED_LIB_EXPORT -Dmimalloc_EXPORTS -IE:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1  /Zc:__cplusplus -std:c++17 /showIncludes /FoCMakeFiles\mimalloc.dir\src\stats.c.obj /FdCMakeFiles\mimalloc.dir\ /FS -c E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\src\stats.c
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(503): error C2975: 'destroy': invalid template argument for '_mi_heap_stl_allocator_common', expected compile-time constant expression
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(484): note: see declaration of 'destroy'
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(519): note: see reference to class template instantiation '_mi_heap_stl_allocator_common<T,destroy>' being compiled
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(514): error C2975: 'destroy': invalid template argument for '_mi_heap_stl_allocator_common', expected compile-time constant expression
E:\thirdsource\vcpkg\buildtrees\mimalloc\src\v2.0.9-b57e26b15c.clean\include\mimalloc.h(484): note: see declaration of 'destroy'
```
**Additional context**

Add any other context about the problem here, such as what you have already tried to resolve the issue.

@JonLiu1993 my environment : Windows11家庭中文版 22H2/vs2019/windowSDK 10.0.19041.0, I have installed the English language pack on my VS2019. Thank you.

```
E:\thirdsource\vcpkg>.\vcpkg.exe install mimalloc --editable
Computing installation plan...
The following packages will be built and installed:
    mimalloc[core]:x86-windows -> 2.0.9
Restored 0 package(s) from C:\Users\70918\AppData\Local\vcpkg\archives in 80.1 us. Use --debug to see more details.
Installing 1/1 mimalloc:x86-windows...
Building mimalloc[core]:x86-windows...
-- Using cached microsoft-mimalloc-v2.0.9.tar.gz.
-- Using source at E:/thirdsource/vcpkg/buildtrees/mimalloc/src/v2.0.9-b57e26b15c
-- Found external ninja('1.10.2').
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: "D:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j21
    Working Directory: E:/thirdsource/vcpkg/buildtrees/mimalloc/x86-windows-dbg
    See logs for more information:
      E:\thirdsource\vcpkg\buildtrees\mimalloc\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/mimalloc/portfile.cmake:36 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)


error: building mimalloc:x86-windows failed with: BUILD_FAILED
internal error: D:\a\_work\1\s\src\vcpkg\build.cpp(1475): Value was null
Please open an issue at https://github.com/microsoft/vcpkg/issues/new?template=other-type-of-bug-report.md&labels=category:vcpkg-bug with detailed steps to reproduce the problem.

```

--- a/ports/mimalloc/portfile.cmake
+++ b/ports/mimalloc/portfile.cmake
@@ -1,8 +1,8 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO microsoft/mimalloc
-    REF v2.0.9
-    SHA512 bf6945bfb600ade35dab34c7f570ee4f69a77612547ad874bbbd989a4e594a6a219c222a22c90c5e36f205aae4d5cd1a5e4651caed5433db275d414c6769bf49
+    REF "v${VERSION}"
+    SHA512 01c7bdfd001ebc34bb0918fe4ecff42a4ec316fbbd73d4dda52c93c31e5712595758a595ae7ea436f445ecc0ebbf7f9c63d9c572f1c1c5a9e96f51fc524a4875
     HEAD_REF master
     PATCHES
         fix-cmake.patch
@@ -55,4 +55,4 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 
 vcpkg_fixup_pkgconfig()
 
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")

--- a/ports/mimalloc/portfile.cmake
+++ b/ports/mimalloc/portfile.cmake
@@ -35,8 +35,6 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
-vcpkg_copy_pdbs()
-
 file(COPY
     "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake"
     DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"

--- a/ports/mimalloc/portfile.cmake
+++ b/ports/mimalloc/portfile.cmake
@@ -35,6 +35,8 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
+vcpkg_copy_pdbs()
+
 file(COPY
     "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake"
     DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"

--- a/ports/mimalloc/vcpkg.json
+++ b/ports/mimalloc/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "mimalloc",
-  "version": "2.0.9",
+  "version": "2.1.1",
   "description": "Compact general purpose allocator with excellent performance",
   "homepage": "https://github.com/microsoft/mimalloc",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5133,7 +5133,7 @@
       "port-version": 6
     },
     "mimalloc": {
-      "baseline": "2.0.9",
+      "baseline": "2.1.1",
       "port-version": 0
     },
     "minc": {

--- a/versions/m-/mimalloc.json
+++ b/versions/m-/mimalloc.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "f6055b5cb8a3a2d744cac4781fed65cf1da1064f",
+      "version": "2.1.1",
+      "port-version": 0
+    },
+    {
       "git-tree": "aaa17dfe033cf63a5884ab214e39a790181ae8a3",
       "version": "2.0.9",
       "port-version": 0

--- a/versions/m-/mimalloc.json
+++ b/versions/m-/mimalloc.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "f6055b5cb8a3a2d744cac4781fed65cf1da1064f",
+      "git-tree": "0120c72c8d2ec48a9210549180e2a4fe209af8a9",
       "version": "2.1.1",
       "port-version": 0
     },

--- a/versions/m-/mimalloc.json
+++ b/versions/m-/mimalloc.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "0120c72c8d2ec48a9210549180e2a4fe209af8a9",
+      "git-tree": "f6055b5cb8a3a2d744cac4781fed65cf1da1064f",
       "version": "2.1.1",
       "port-version": 0
     },

